app/src/main/java/net/gsantner/opoc/util/Callback.java

```diff
@@ -84,4 +84,24 @@
     public interface s5<A, B, C, D, E> {
         String callback(A arg1, B arg2, C arg3, D arg4, E arg5);
     }
+
+    public interface r1<R, A> {
+        R callback(A arg1);
+    }
+
+    public interface r2<R, A, B> {
+        R callback(A arg1, B arg2);
+    }
+
+    public interface r3<R, A, B, C> {
+        R callback(A arg1, B arg2, C arg3);
+    }
+
+    public interface r4<R, A, B, C, D> {
+        R callback(A arg1, B arg2, C arg3, D arg4);
+    }
+
+    public interface r5<R, A, B, C, D, E> {
+        R callback(A arg1, B arg2, C arg3, D arg4, E arg5);
```

Comment:
We could replace all the other callback types with this if we wanted to.

Reply:
Think it's ok as is.

Reply:
Think it's ok as is.

There is a big plus with the old ones: Impossible null values. boolean/int/long can't be null. With generics you can only have Boolean/Integer